packages/trader/src/Stores/client-store.js

```diff
@@ -975,6 +980,29 @@ export default class ClientStore extends BaseStore {
         this.device_data = device_data;
     }
 
+    @action.bound
+    onSetResidence({ residence }, cb) {
+        if (!residence) return;
+        WS.setSettings({ residence: residence }).then(async response => {
+            if (response.error) {
+                cb(response.error.message);
+            } else {
+                await this.setResidence(residence);
+                await WS.authorized.storage.landingCompany(this.accounts[this.loginid].residence)
+                    .then(this.responseLandingCompany);
+                await WS.authorized.storage.getSettings().then(async response => {
```

Comment:
since (I think) we always need to get settings after set settings we could have a separate function for this?

Reply:
Yeah, we're doing this as well currently in Personal Details component in Accounts Settings. We're doing some state setting as well there with the response (in Personal Details) whereas here we're just updating the `observable` in `client-store`.